lib/spack/spack/stage.py

```diff
@@ -44,6 +44,14 @@
 stage_prefix = 'spack-stage-'
 
 
+def ensure_external_stage_path(stage_root):
+    # type: (str) -> None
+    """Ensure that a directory structure exists for a user-specified stage path.
+
+    This is used when running `spack stage -p <PATH>`."""
+    mkdirp(os.path.join(stage_root, _source_path_subdir))
```

Comment:
Is this method necessary? My instinct is that we should be using `_create_stage_root` here (and maybe renaming it to `create_stage_root` if necessary -- I don't think there's anything  in it that absolutely needs  to be a private method). Users of a shared Spack instance could create a stage in a shared directory -- I think we want the permissions-handling magic from the existing stage logic.

Reply:
Ok, that was my thought but wasn't sure. I'll make this change.

Reply:
There's a very weird issue where using the `create_stage_root()` method will *not* extract the downloaded archive into `spack-src`, while using `mkdirp()` directly will. I agree that it seems like this method should be used for this purpose, so I will dig into why that is occurring.

Reply:
I am finding that the stage *does* get correctly downloaded into the specified directory, and it seems like whether it gets extracted is not influenced by how we use `mkdirp()` here, so I have made the exact change you describe!